thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java

```diff
@@ -249,6 +249,13 @@ public DetectionPipelineResult run() throws Exception {
       checkEarlyStop(totalNestedMetrics, successNestedMetrics, i, lastException);
       MetricEntity metric = nestedMetrics.get(i);
       try {
+        if (Thread.interrupted()) {
+          // this happens when Future.cancel() or ExecutorService.shutdown() is called.
+          LOG.error("Current thread is interrupted before running dimension {}/{} for detection {} metrics {}, "
+                  + "so skip the rest of dimensions and return the intermediate",
+              i + 1 , totalNestedMetrics, this.config.getId(), metric.getUrn());
+          break;
```

Comment:
Add a comment: "We still want to keep the current detection result and store it anyway."